src/game.h

```diff
@@ -644,7 +645,8 @@ class game
          */
         bool handle_liquid( item &liquid, item *source = NULL, int radius = 0,
                             const tripoint *source_pos = nullptr,
-                            const vehicle *source_veh = nullptr );
+                            const vehicle *source_veh = nullptr,
+                            const monster *source_mon = nullptr );
```

Comment:
This function is growing awfully huge and multi-purpose.
5 optional arguments, 4 of which default to `null` is a good sign that this will need a refactor soon.
That extra argument is pushing it, but could be tolerated for now.